code/modules/cm_marines/dropship_ammo.dm

```diff
@@ -292,6 +292,97 @@
+	/// the maximum volume allowed in the rocket
+	var/max_container_volume = 300
+	/// the current total volume
+	var/current_container_volume = 0
+	/// the containers inside the rocket
+	var/list/obj/item/reagent_container/glass/containers = list()
+	var/list/allowed_containers = list(/obj/item/reagent_container/glass/beaker, /obj/item/reagent_container/glass/bucket, /obj/item/reagent_container/glass/bottle, /obj/item/reagent_container/glass/bucket/mopbucket)
+
+/obj/structure/ship_ammo/rocket/custom_missile/attackby(obj/item/item as obj, mob/user as mob)
+	if(HAS_TRAIT(item, TRAIT_TOOL_SCREWDRIVER))
+		if(assembly_stage == ASSEMBLY_UNLOCKED)
+			to_chat(user, SPAN_NOTICE("You lock the [name]."))
+			playsound(loc, 'sound/items/Screwdriver.ogg', 25, 0, 6)
+			assembly_stage = ASSEMBLY_LOCKED
+			icon_state = "custom_missile_ready"
+		else if(assembly_stage == ASSEMBLY_LOCKED)
+			to_chat(user, SPAN_NOTICE("You unlock the [name]."))
+			playsound(loc, 'sound/items/Screwdriver.ogg', 25, 0, 6)
+			assembly_stage = ASSEMBLY_UNLOCKED
+			icon_state = "custom_missile"
+	else if(is_type_in_list(item, allowed_containers) && (!assembly_stage || assembly_stage == ASSEMBLY_UNLOCKED))
+		if(current_container_volume >= max_container_volume)
+			to_chat(user, SPAN_DANGER("The [name] can not hold more containers."))
+			return
+		else
+			if(item.reagents.total_volume)
+				if(item.reagents.maximum_volume + current_container_volume > max_container_volume)
+					to_chat(user, SPAN_DANGER("\the [item] is too large for [name]."))
+					return
+				if(user.temp_drop_inv_item(item))
+					to_chat(user, SPAN_NOTICE("You add \the [item] to the [name]."))
+					item.forceMove(src)
+					containers += item
+					current_container_volume += item.reagents.maximum_volume
+					assembly_stage = ASSEMBLY_UNLOCKED
+			else
+				to_chat(user, SPAN_DANGER("\the [item] is empty."))
+	else if(HAS_TRAIT(item, TRAIT_TOOL_CROWBAR))
+		if(assembly_stage == ASSEMBLY_UNLOCKED)
+			if(containers.len)
+				for(var/obj/container in containers)
+					if(istype(container))
+						containers -= container
+						user.put_in_hands(container)
+				current_container_volume = 0
+	else return ..()
+
+/obj/structure/ship_ammo/rocket/custom_missile/get_examine_text(mob/user)
+	. = ..()
+	. += "\n Contains [containers.len] containers."
+	switch(assembly_stage)
+		if(ASSEMBLY_LOCKED)
+			. += "\n It is ready."
+		if(ASSEMBLY_UNLOCKED)
+			. += "\n It is unlocked."
+
+/obj/structure/ship_ammo/rocket/custom_missile/detonate_on(turf/impact, obj/structure/dropship_equipment/weapon/fired_from)
+	if(assembly_stage == ASSEMBLY_UNLOCKED || containers.len == 0) //shitty explosion if left unlocked or no containers
+		impact.ceiling_debris_check(3)
+		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), impact, 60, 30, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, create_cause_data(initial(name)), source_mob), 0.5 SECONDS)
+	else
+		if(assembly_stage == ASSEMBLY_LOCKED)
```

Comment:
you can probably move up the if to make else if.  or just an if works too probably